### PR TITLE
bug(server): set connection flags block/pause flag on all blocking commands

### LINKED ISSUE #2660: Bug: set connection blocking flag on all blocking commands
need to set blocking flag inside Transaction:: WaitOnWatch
Today we set the blocking connection flag only from RunCbOnFirstNonEmptyBlocking

### PR DESCRIPTION
**Pull request by adiholden**
fixes #2660 
The bug:
connection blocking and pause flags are not set on all blocking commands.
Therefor if client pause is running when running brpoplpush it will not be able to puase and return timeout, as
the DispatchTracker will not know that this connections are blocking connecitons
The fix:
Pass connection cntx to WaitOnWatch to set the blocking and pause flags

### CODE REVIEW

File: src/server/transaction.h [hunk 1]
```diff
@@ -213,7 +213,8 @@ class Transaction {
   // or b) tp is reached. If tp is time_point::max() then waits indefinitely.
   // Expects that the transaction had been scheduled before, and uses Execute(.., true) to register.
   // Returns false if timeout occurred, true if was notified by one of the keys.
-  facade::OpStatus WaitOnWatch(const time_point& tp, WaitKeysProvider cb, KeyReadyChecker krc);
+  facade::OpStatus WaitOnWatch(const time_point& tp, WaitKeysProvider cb, KeyReadyChecker krc,
+                               ConnectionContext* cntx);
```
Comment (dranikpg) [inline src/server/transaction.h:217]: In the future I'd suggest not passing ConnectionContext* but instead a FlagPointers { bool* paused, bool* blocked }, so we don't have this general type here in transaction code. Just a nit here, I can update it later

File: src/server/container_utils.cc [hunk 2]
```diff
@@ -258,7 +258,7 @@ string_view LpGetView(uint8_t* lp_it, uint8_t int_buf[]) {
 
 OpResult<string> RunCbOnFirstNonEmptyBlocking(Transaction* trans, int req_obj_type,
                                               BlockingResultCb func, unsigned limit_ms,
-                                              bool* block_flag) {
+                                              ConnectionContext* cntx) {
```
Comment (dranikpg) [inline src/server/container_utils.cc:261]: Like I did it here 😅

**Review by dranikpg — APPROVED** [review]
I'm not sure why it fails

```
>       assert res == ["dest", "value"]
E       AssertionError: assert equals failed
E         None        [          
E                       'dest',  
E                       'value', 
E                     ]
```

**Comment by adiholden** [comment]
> I'm not sure why it fails
> 
> ```
> >       assert res == ["dest", "value"]
> E       AssertionError: assert equals failed
> E         None        [          
> E                       'dest',  
> E                       'value', 
> E                     ]
> ```

I will open a separate issue for this there is bug in brpoplpush when running on single shard

**Comment by adiholden** [comment]
@dranikpg please review

**Review by dranikpg — APPROVED** [review]
Sorry 😅

**Pull request merged**